spyql/processor.py

```diff
@@ -44,38 +45,56 @@ def init_vars():
         spyql.log.user_debug(f"Init file not found: {init_fname}")
     except Exception as e:
         spyql.log.user_warning(f"Could not load {init_fname}", e)
+    finally:
+        # user defined variables
+        vars["user_query_vars"] = user_query_vars
     return vars
 
 
 class Processor:
     @staticmethod
-    def make_processor(prs, strings, input_options):
+    def make_processor(prs, strings, interactive = False, input_options = {}):
         """
         Factory for making a file processor based on the parsed query
         """
+
+        spyql.log.user_info(f"Input Options: {input_options}")
+
+        # Since I did not want to damage the existing flow using FROM, when the query
+        # is *really* interactive (as determined by Q), we set "interactive" in the
+        # parsed dict. This is different from interactive passed in the spec above,
+        # which tells the processors that you have to read data from filepath and not
+        # from stdin.
+        if prs.get("interactive", None):
+            return InteractiveProcessor(prs=prs, strings=strings, **input_options)
+
         try:
+
             processor_name = prs["from"]
             if not processor_name:
-                return Processor(prs, strings, **input_options)
+                return Processor(prs=prs, strings=strings, interactive=interactive, **input_options)
 
             processor_name = processor_name.upper()
 
             if processor_name == "JSON":
-                return JSONProcessor(prs, strings, **input_options)
-            if processor_name == "CSV":
-                return CSVProcessor(prs, strings, **input_options)
-            if processor_name == "TEXT":  # single col
-                return TextProcessor(prs, strings, **input_options)
-            if processor_name == "SPY":
-                return SpyProcessor(prs, strings, **input_options)
-
-            return PythonExprProcessor(prs, strings, **input_options)
+                return JSONProcessor(prs=prs, strings=strings, interactive=interactive, **input_options)
+            elif processor_name == "CSV":
+                return CSVProcessor(prs=prs, strings=strings, interactive=interactive, **input_options)
+            elif processor_name == "TEXT":  # single col
+                return TextProcessor(prs=prs, strings=strings, interactive=interactive, **input_options)
+            elif processor_name == "SPY":
+                return SpyProcessor(prs=prs, strings=strings, interactive=interactive, **input_options)
+
+            return PythonExprProcessor(prs=prs, strings=strings, interactive=interactive, **input_options)
         except TypeError as e:
             spyql.log.user_error(f"Could not create '{processor_name}' processor", e)
 
-    def __init__(self, prs, strings):
+    def __init__(self, prs, strings, interactive = False):
+        spyql.log.user_info(f"Loading {self.__class__.__name__}")
         self.prs = prs  # parsed query
+        spyql.log.user_info(self.prs)
```

Comment:
```suggestion
        spyql.log.user_dbg(f"Loading {self.__class__.__name__}")
        self.prs = prs  # parsed query
        spyql.log.user_dbg(self.prs)
```
I prefer leaving this kind of internal outputs to debug level... I believe `prs` is already being outputted in the CLI, so moving to here might make sense.